api/prisma/schema.prisma

```diff
@@ -375,15 +376,13 @@ model UserToken {
   @@index([userId], map: "userId_1")
 }
 
-/// TODO: Token has to outlive the exam attempt
-/// Validation has to be taken as the attempt is requested
-/// to ensure it lives long enough.
 model ExamEnvironmentAuthorizationToken {
   id          String   @id @map("_id")
   createdDate DateTime @db.Date
-  userId      String   @db.ObjectId
+  userId      String   @unique @db.ObjectId
```

Comment:
In order to impose this constraint we should `prisma db push` or create the constraint by hand. That applies to the prod db as well as the tests.

Reply:
> Why not not include this in the user object?

Include what in the `user` object? This PR includes the token in the `user` object.

> In order to impose this constraint we should prisma db push or create the constraint by hand.

What are you saying? This is a new collection (does not exist yet). So, the constraint cannot cause issues?

Reply:
> Include what in the user object? This PR includes the token in the user object.

Yeah, that. There's already an endpoint to get the token, so why put it in the user object?

> What are you saying? This is a new collection (does not exist yet). So, the constraint cannot cause issues?

That adding the constraint to the schema does nothing on its own - it has to be applied to the dbs (e.g. via `prisma db push`) to take effet.

Reply:
> There's already an endpoint to get the token, so why put it in the user object?

Oh. Sorry. My brain ignored the second _"not"_ in your sentence.

Reason: There is no endpoint to get the token. There is only an endpoint to generate it.

Would you just create an endpoint to GET/generate it?

Reply:
>  it has to be applied to the dbs

Ah. I thought you were suggesting we needed a migration to use the constraint.

Reply:
Very true. Thank you.